### PR TITLE
Add a new Waffle tutorial

### PR DESCRIPTION
**Pull request by MiZiet**
## Description

Add a new Hello World tutorial for Waffle.

### CODE REVIEW

**Comment by samajammin** [comment]
Thanks for the submission @MiZiet! I've converted this PR to a draft. Please [mark it as ready for review](https://docs.github.com/en/free-pro-team@latest/github/collaborating-with-issues-and-pull-requests/changing-the-stage-of-a-pull-request#marking-a-pull-request-as-ready-for-review) when it's no longer a WIP.

**Comment by samajammin** [comment]
Hey @MiZiet - checking in on this.

**Comment by samajammin** [comment]
Hey @MiZiet - another nudge! We'd like to get this merged. Please let me know if/when this is ready to review. Thanks.

**Comment by MiZiet** [comment]
Hey, sorry that it took me so long.

**Review by samajammin — CHANGES_REQUESTED** [review]
Thanks again for the submission @MiZiet. Given that Buidler is now Hardhat, would you please update the tutorial to reflect that?

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+lang: en
+sidebar: true
+published: 2020-10-16
+---
+
+### In this [Waffle](https://ethereum-waffle.readthedocs.io) tutorial, we will learn how to set up a simple "Hello world" smart contract project, using [hardhat](https://hardhat.org/) and [ethers.js](https://docs.ethers.io/v5/). Then we will learn how to add a new functionality to our smart contract and how to test it with Waffle.
+
+Let's start with creating new project:
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:17]:
```suggestion
```bash
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+yarn init
+```
+
+or
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:23]:
```suggestion
```bash
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+npm init
+```
+
+and installing required packages:
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:29]:
```suggestion
```bash
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 4]
```diff
@@ -0,0 +1,196 @@
...
+yarn add -D hardhat @nomiclabs/hardhat-ethers ethers @nomiclabs/hardhat-waffle ethereum-waffle chai
+```
+
+or
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:35]:
```suggestion
```bash
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 5]
```diff
@@ -0,0 +1,196 @@
...
+npm install -D hardhat @nomiclabs/hardhat-ethers ethers @nomiclabs/hardhat-waffle ethereum-waffle chai
+```
+
+Next step is creating a sample hardhat project by running `npx hardhat`.
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:41]:
```suggestion
```bash
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 6]
```diff
@@ -0,0 +1,196 @@
...
+├── .gitignore
+├── hardhat.config.js
+└── package.json
+```
+
+### Now let's talk about some of these files:
+
+- Greeter.sol - our smart contract writen in solidity;
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:82]:
```suggestion
```solidity
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 7]
```diff
@@ -0,0 +1,196 @@
...
+
+Our smart contract can be divided into three parts:
+
+1. constructor - where we declare a string type variable called `greeting`,
+2. function greet - a function that will return the `greeting` when called,
+3. function setGreeting - a function that allows us to change the `greeting` value.
+
+- sample-test.js - our tests file
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:110]:
```suggestion
```js
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 8]
```diff
@@ -0,0 +1,196 @@
...
+expect(await greeter.greet()).to.equal("Hola, mundo!");
+});
+});
+```
+
+### Next step consists in compiling our contract and runing tests:
+
+Waffle tests use Mocha (a test framework) with Chai (an assertion library). All you have to do is run `npx hardhat test` and wait for the following message to appear.
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:129]:
```suggestion
```bash
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+```
+
+### Everything looks great so far, let's add some more complexity to our project :)
+
+Imagine a situation when someone adds an empty string as a greeting. It wouldn't be a warm greeting, right?  
+Let's make sure that doesn't happen:
+
+We want to use solidity's `revert` when someone passes an empty string. A good thing is that we can easily test this functionality with Waffle's chai matcher `to.bo.revertedWith()`.
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:140]:
```suggestion
```js
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 10]
```diff
@@ -0,0 +1,196 @@
...
+const greeter = await Greeter.deploy("Hello, world!");
+
+await greeter.deployed();
+await expect(greeter.setGreeting("")).to.be.revertedWith('Greeting should not be empty');
+});
+```
+
+Looks like our new test didn't pass:
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:152]:
```suggestion
```bash
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 11]
```diff
@@ -0,0 +1,196 @@
...
+    1) Should revert when passing an empty string
+
+
+  1 passing (2s)
+  1 failing
+```
+
+Let's implement this functionality into our smart contract:
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:167]:
```suggestion
```solidity
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 12]
```diff
@@ -0,0 +1,196 @@
...
+require(bytes(_greeting).length > 0, "Greeting message is empty");
+```
+
+Now, our setGreeting function looks like this:
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:173]:
```suggestion
```solidity
```

File: src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md [hunk 13]
```diff
@@ -0,0 +1,196 @@
...
+function setGreeting(string memory _greeting) public {
+require(bytes(_greeting).length > 0, "Greeting should not be empty");
+console.log("Changing greeting from '%s' to '%s'", greeting, _greeting);
+greeting = _greeting;
+}
+```
+
+Let's run tests again:
+
+```yml
```
Comment (samajammin) [inline src/content/developers/tutorials/Waffle-hello-world-with-buidler-tutorial/index.md:183]:
```suggestion
```bash
```

**Review by samajammin — APPROVED** [review]
Thanks for making these updates! Approving w/ syntax tweaks.

**Review by samajammin — APPROVED** [review]

**Comment by samajammin** [comment]
@all-contributors please add @MiZiet for content.

**Pull request merged**